# HG changeset patch
# User Rik <rik@octave.org>
# Date 1512595155 28800
#      Wed Dec 06 13:19:15 2017 -0800
# Node ID 77d976a84d0adf6336f16fef37e14c37908e182b
# Parent  1909f2d7a36eee6cb281ca9def2440f006709d31
# Parent  ccedd8a29d3e14ae704732176863e1d616bb401c
maint: merge stable to default.

diff --git a/scripts/plot/util/private/__gnuplot_draw_axes__.m b/scripts/plot/util/private/__gnuplot_draw_axes__.m
--- a/scripts/plot/util/private/__gnuplot_draw_axes__.m
+++ b/scripts/plot/util/private/__gnuplot_draw_axes__.m
@@ -951,17 +951,17 @@ function __gnuplot_draw_axes__ (h, plot_
                     endif
                     color = "flat";
                     have_cdata(data_idx) = true;
                   endif
                 elseif (strcmp (ec, "interp"))
                   if (do_interp_warning && numel (ccol) == 3)
                     warning ('"interp" not supported, using 1st entry of cdata');
                     do_interp_warning = false;
-                    color = ccol(1,:);
+                    color = mapcdata (ccol(:), obj.cdatamapping, clim, cmap_sz);
                   else
                     if (isscalar (ccol))
                       ccol = repmat (ccol, numel (xcol), 1);
                     endif
                     color = "interp";
                     have_cdata(data_idx) = true;
                   endif
                   ccol = mapcdata (ccol, obj.cdatamapping, clim, cmap_sz);
